File: 9041/test06/equi_filter.py
#!/usr/bin/python3
import sys

for line in sys.stdin:
    line = line.strip()
    for word in line.split(' '):
        sts = True
        d = {}
        word2 = word.lower()
        for char in word2:
            if char in d:
                d[char] += 1
            else:
                d[char] = 1
        a = word2[0]
        value = d[a]
        for key in d:
            if d[key] != value:
                sts = False
                break
        if sts:
            print(word, end=" ")
    print()
    
    
    
        
            

File: 9041/test06/equi_filter.pl
#!/usr/bin/perl -w


while (my $line = <STDIN>) {
	my @words = split(/\s/, $line);
	foreach my $word (@words) {
		my $sts = 1;
		my %dict = ();
		foreach my $char (split(//, $word)) {
			if ($dict{lc $char}) {
				$dict{lc $char}++;
			} else {
				$dict{lc $char} = 1;
			}
		}
		my $a = lc substr($word, 0, 1);
		my $value = $dict{$a};
		foreach my $key (keys %dict) {
			if ($dict{$key} != $value) {
				$sts = 0;
				last;
			}
		}
		if ($sts == 1) {
			print "$word ";
		}
	}
	print "\n";
	
}
			
			
		
